### PR TITLE
rails 6.3.0.1

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
@tbrisker here ya go

@evgeni @zjhuntin can you review? I can drop the rails_core_packages group if y'all prefer. It came in very handy for doing these sorta updates.

File: package_manifest.yaml [hunk 1]
```diff
@@ -70,19 +88,6 @@ foreman_scl_packages:
...
     gyp: {}
-    rubygem-actioncable: {}
-    rubygem-actionmailbox: {}
-    rubygem-actionmailer: {}
-    rubygem-actiontext: {}
-    rubygem-activejob: {}
-    rubygem-activemodel: {}
-    rubygem-activerecord: {}
-    rubygem-activerecord-nulldb-adapter: {}
-    rubygem-activerecord-session_store: {}
```
Comment (evgeni) [inline package_manifest.yaml:81]: those two somehow got lost?!

File: packages/foreman/rubygem-actioncable/rubygem-actioncable.spec [hunk 2]
```diff
@@ -3,33 +3,32 @@
...
 Requires: %{?scl_prefix}rubygem(websocket-driver) >= 0.6.1
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-actioncable/rubygem-actioncable.spec:31]: IMHO the obsoletes should still be there

File: packages/foreman/rubygem-actioncable/rubygem-actioncable.spec [hunk 3]
```diff
@@ -3,33 +3,32 @@
...
-Requires: %{?scl_prefix}rubygem(actionpack) = 6.0.2.2
+Requires: %{?scl_prefix_ruby}ruby(rubygems) 
+Requires: %{?scl_prefix}rubygem(actionpack) = 6.0.3.1
 Requires: %{?scl_prefix}rubygem(nio4r) >= 2.0
 Requires: %{?scl_prefix}rubygem(nio4r) < 3
 Requires: %{?scl_prefix}rubygem(websocket-driver) >= 0.6.1
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
```
Comment (evgeni) [inline packages/foreman/rubygem-actioncable/rubygem-actioncable.spec:27]: any reason there is a trailing space now?

File: packages/foreman/rubygem-actioncable/rubygem-actioncable.spec [hunk 4]
```diff
@@ -3,33 +3,32 @@
...
-URL: http://rubyonrails.org
+URL: https://rubyonrails.org
 Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
 
 # start specfile generated dependencies
 Requires: %{?scl_prefix_ruby}ruby(release)
 Requires: %{?scl_prefix_ruby}ruby >= 2.5.0
-Requires: %{?scl_prefix_ruby}ruby(rubygems)
-Requires: %{?scl_prefix}rubygem(actionpack) = 6.0.2.2
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
```
Comment (evgeni) [inline packages/foreman/rubygem-actioncable/rubygem-actioncable.spec:20]: space?

File: packages/foreman/rubygem-actionmailbox/rubygem-actionmailbox.spec [hunk 5]
```diff
@@ -16,16 +17,16 @@ Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
...
+Requires: %{?scl_prefix}rubygem(activejob) = 6.0.3.1
+Requires: %{?scl_prefix}rubygem(actionpack) = 6.0.3.1
 Requires: %{?scl_prefix}rubygem(mail) >= 2.7.1
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
```
Comment (evgeni) [inline packages/foreman/rubygem-actionmailbox/rubygem-actionmailbox.spec:32]: I wonder why there is no obsoletes here, is mailbox new in 6.0?

File: packages/foreman/rubygem-actionmailer/rubygem-actionmailer.spec [hunk 6]
```diff
@@ -3,39 +3,38 @@
...
 Requires: %{?scl_prefix}rubygem(rails-dom-testing) < 3
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-actionmailer/rubygem-actionmailer.spec:35]: please keep

File: packages/foreman/rubygem-actionpack/rubygem-actionpack.spec [hunk 7]
```diff
@@ -3,39 +3,39 @@
...
+Requires: %{?scl_prefix}rubygem(actionview) = 6.0.3.1
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-actionpack/rubygem-actionpack.spec:37]: please keep

File: packages/foreman/rubygem-actiontext/rubygem-actiontext.spec [hunk 8]
```diff
@@ -16,15 +17,15 @@ Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
...
+Requires: %{?scl_prefix}rubygem(activestorage) = 6.0.3.1
+Requires: %{?scl_prefix}rubygem(actionpack) = 6.0.3.1
 Requires: %{?scl_prefix}rubygem(nokogiri) >= 1.8.5
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
```
Comment (evgeni) [inline packages/foreman/rubygem-actiontext/rubygem-actiontext.spec:31]: no obsoletes here?

File: packages/foreman/rubygem-actionview/rubygem-actionview.spec [hunk 9]
```diff
@@ -29,13 +30,11 @@ Requires: %{?scl_prefix}rubygem(rails-dom-testing) >= 2.0
 Requires: %{?scl_prefix}rubygem(rails-dom-testing) < 3
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-actionview/rubygem-actionview.spec:37]: please keep

File: packages/foreman/rubygem-activejob/rubygem-activejob.spec [hunk 10]
```diff
@@ -3,33 +3,32 @@
...
 Requires: %{?scl_prefix}rubygem(globalid) >= 0.3.6
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-activejob/rubygem-activejob.spec:29]: please keep

File: packages/foreman/rubygem-activemodel/rubygem-activemodel.spec [hunk 11]
```diff
@@ -3,30 +3,29 @@
...
+Requires: %{?scl_prefix}rubygem(activesupport) = 6.0.3.1
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-activemodel/rubygem-activemodel.spec:28]: please keep

File: packages/foreman/rubygem-activerecord/rubygem-activerecord.spec [hunk 12]
```diff
@@ -3,31 +3,30 @@
...
+Requires: %{?scl_prefix}rubygem(activemodel) = 6.0.3.1
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-activerecord/rubygem-activerecord.spec:29]: please keep

File: packages/foreman/rubygem-activestorage/rubygem-activestorage.spec [hunk 13]
```diff
@@ -3,34 +3,33 @@
...
 Requires: %{?scl_prefix}rubygem(marcel) < 0.4
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-activestorage/rubygem-activestorage.spec:32]: please keep

File: packages/foreman/rubygem-activesupport/rubygem-activesupport.spec [hunk 14]
```diff
@@ -28,15 +29,14 @@ Requires: %{?scl_prefix}rubygem(concurrent-ruby) < 2
...
+Requires: %{?scl_prefix}rubygem(zeitwerk) >= 2.2.2
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-activesupport/rubygem-activesupport.spec:38]: please keep

File: packages/foreman/rubygem-rails/rubygem-rails.spec [hunk 15]
```diff
@@ -38,8 +39,6 @@ BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-rails/rubygem-rails.spec:41]: please keep

File: packages/foreman/rubygem-railties/rubygem-railties.spec [hunk 16]
```diff
@@ -3,35 +3,34 @@
...
+Requires: %{?scl_prefix}rubygem(method_source) 
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby >= 2.5.0
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel 
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 # end specfile generated dependencies
 
-Obsoletes: tfm-ror52-rubygem-%{gem_name} <= 5.2.1
```
Comment (evgeni) [inline packages/foreman/rubygem-railties/rubygem-railties.spec:33]: please keep

**Comment by evgeni** [comment]
the manifest change is cool!

File: packages/foreman/rubygem-actioncable/rubygem-actioncable.spec (hunk 2 above)
Comment (ehelms) [reply packages/foreman/rubygem-actioncable/rubygem-actioncable.spec:31]: Silly non-idempotent spec updates :)

File: packages/foreman/rubygem-actioncable/rubygem-actioncable.spec (hunk 3 above)
Comment (ehelms) [reply packages/foreman/rubygem-actioncable/rubygem-actioncable.spec:27]: Gem2rpm said so? Will remove

File: packages/foreman/rubygem-actionmailbox/rubygem-actionmailbox.spec (hunk 5 above)
Comment (ehelms) [reply packages/foreman/rubygem-actionmailbox/rubygem-actionmailbox.spec:32]: Aye

File: packages/foreman/rubygem-actiontext/rubygem-actiontext.spec (hunk 8 above)
Comment (ehelms) [reply packages/foreman/rubygem-actiontext/rubygem-actiontext.spec]: New package with 6.Y

**Comment by ehelms** [comment]
I was able to make this update with:

```
obal update rails_core_packages --version 6.0.3.1
obal update-spec rails_core_packages --template gem2rpm/scl.spec.erb
```

These relied upon:

https://github.com/theforeman/obal/pull/207
https://github.com/theforeman/obal/pull/258

**Comment by evgeni** [comment]
@ehelms can you please open a CP agains 2.1? I'll do the tagging dance.

**Pull request merged**